### DISCUSSION TITLE
Problems creating new Permissions

### DISCUSSION THREAD

**Discussion by khanhduatd186** [abpframework/abp#20206] [Q&A]
when running debug, the above code can run through all but cannot get group as well as AddGroup and AddPermission, make sure access is the highest permission, please help me

namespace HQSOFT.eBiz.CoreBackend.SalesPriceCustomerPriceClasses
{
    public class ReportCreatedEventHandler : ILocalEventHandler<EntityCreatedEventData<SalesPriceCustomerPriceClass>>,ITransientDependency
    {

        private readonly IPermissionDefinitionContext _context;

        public ReportCreatedEventHandler(IPermissionDefinitionContext context)
        {
            _context = context;

        }
        public Task HandleEventAsync(EntityCreatedEventData<SalesPriceCustomerPriceClass> eventData)
        {
            try
            {
                var reportName = eventData.Entity.CustomerPriceClassCode;
                var permissionName = $"Reports.{reportName}";
                var group =  _context.GetGroupOrNull("Reports");
                var group1 = _context.GetGroupOrNull("CoreBackend");
                Console.WriteLine("checkGroup: ", group.Name, group1.Name);
                if (group == null)
                {
                    var myGroup = _context.AddGroup("Reports", L("Permission:Reports"));
                    if (myGroup != null)
                    {
                        myGroup.AddPermission(permissionName, L($"Permission:{reportName}"));
                    }
                }
                else
                {
                    var permission = group.GetPermissionOrNull(permissionName);
                    if (permission == null)
                    {
                        group.AddPermission(permissionName, L($"Permission:{reportName}"));
                    }
                }
            }
            catch
            {

            }

            return Task.CompletedTask;
        }

        private static LocalizableString L(string name)
        {
            return LocalizableString.Create<CoreBackendResource>(name);
        }

    }
}

**Comment by maliming** [comment]
hi

You can check this question: https://support.abp.io/QA/Questions/5490/Need-to-refresh-the-list-of-static-permission-definitions-without-site-reload
